aten/src/ATen/native/ReflectionPad.cpp

```diff
@@ -81,14 +77,7 @@ TORCH_META_FUNC(reflection_pad1d)(const Tensor& input, IntArrayRef padding) {
 TORCH_META_FUNC(reflection_pad1d_backward)(const Tensor& grad_output,
     const Tensor& input,
     IntArrayRef padding) {
-  int64_t dim_w = 1;
-  int64_t nbatch = 1;
-
-  if (input.ndimension() == 3) {
```

Comment:
Hm, how come this was possible to be removed?

Reply:
The thing is since we directly use a kernel to implement `reflection_pad1d_backward`, we don't need to get `nbatch` any more. Just need to get `dim_w` in order to check the `input_w`.

We may have two types inputs:
* batch mode: NCW
* non batch mode: CW
 
in either type, `dim_w` is the last dimension, so we can simplify the code with a single line of
`int64_t dim_w = input.dim() - 1;`

Reply:
Hm, my only worry is that this logic will break down if there's no dimension check. 

I know the documentation says there's only these two cases, but from what I can tell the code doesn't enforce it. 

Adding explicit checks also doesn't seem to be worth the trouble.

Can we just keep the current logic? I don't think cleaning this up is worth the potential fallout.

Reply:
@mingfeima - This is the comment I meant that's left to resolve.

Reply:
Oh sure, the PR is updated!